modules/stochastic_tools/src/actions/StochasticResultsAction.C

```diff
@@ -0,0 +1,62 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "StochasticResultsAction.h"
+#include "ActionWarehouse.h"
+#include "ActionFactory.h"
+#include "Transfer.h"
+#include "SetupMeshAction.h"
+#include "SamplerPostprocessorTransfer.h"
+#include "StochasticResults.h"
+
+registerMooseAction("StochasticToolsApp",
+                    StochasticResultsAction,
+                    "declare_stochastic_results_vectors");
+
+InputParameters
+StochasticResultsAction::validParams()
+{
+  InputParameters params = Action::validParams();
+  params.addClassDescription("Action for performing initialization of StochasticResults vectors "
+                             "based on SamplerPostprocessorTransfer.");
+  return params;
+}
+
+StochasticResultsAction::StochasticResultsAction(InputParameters params) : Action(params) {}
+
+void
+StochasticResultsAction::act()
+{
+  if (_current_task == "declare_stochastic_results_vectors")
+  {
+    for (std::shared_ptr<Transfer> & transfer_ptr :
+         _problem->getTransfers(Transfer::DIRECTION::FROM_MULTIAPP))
+    {
+      std::shared_ptr<SamplerPostprocessorTransfer> ptr =
```

Comment:
I know you hate `auto` (and I am in the middle on it) but this an example where the RHS is explicitly specifying the type. Mostly just saves you some typing to use `auto` here

Reply:
I can get on `auto` on the left if the type is on the right. I will try to make it more consistent.